tst/EnergyPlus/unit/SingleDuct.unit.cc

```diff
@@ -0,0 +1,404 @@
+
+		DataGlobals::NumOfTimeStepInHour = 1; // must initialize this to get schedules initialized
+		DataGlobals::MinutesPerTimeStep = 60; // must initialize this to get schedules initialized
+		ScheduleManager::ProcessScheduleInput(); // read schedules
+		ScheduleManager::ScheduleInputProcessed = true;
+		DataEnvironment::Month = 1;
+		DataEnvironment::DayOfMonth = 21;
+		DataGlobals::HourOfDay = 1;
+		DataGlobals::TimeStep = 1;
+		DataEnvironment::DSTIndicator = 0;
+		DataEnvironment::DayOfWeek = 2;
+		DataEnvironment::HolidayIndex = 0;
+		DataEnvironment::DayOfYear_Schedule = General::JulianDay(DataEnvironment::Month, DataEnvironment::DayOfMonth, 1);
+		DataEnvironment::StdRhoAir = Psychrometrics::PsyRhoAirFnPbTdbW( 101325.0, 20.0, 0.0 ); 
+		ScheduleManager::UpdateScheduleValues();
+
+		bool ErrorsFound = false;
+		HeatBalanceManager::GetZoneData(ErrorsFound);
+		ASSERT_FALSE(ErrorsFound);
+		DataZoneEquipment::GetZoneEquipmentData1();
+		ZoneAirLoopEquipmentManager::GetZoneAirLoopEquipment();
+		SingleDuct::GetSysInput();
+		EXPECT_TRUE( compare_err_stream( "" ) );
+		DataHeatBalFanSys::TempControlType.allocate( 1 );
+		DataHeatBalFanSys::TempControlType( 1 ) = DataHVACGlobals::DualSetPointWithDeadBand;
+
+		// node number table
+		//  1   Zone 1 Air Node
+		//  2   Zone 1 Return Node
+		//  3   Zone 1 Supply Inlet
+		//  4   Zone 1 Zone Equip Inlet
+
+		DataZoneEnergyDemands::ZoneSysEnergyDemand.allocate(1);
+		DataZoneEnergyDemands::ZoneSysEnergyDemand( 1 ).RemainingOutputRequired = -2000.0; // Zone 2 has a cooling load
+
+		// Setup for Zone 1 VAV No Reheat
+		int SysNum = 1;
+		int ZoneNum = 1;
+		int ZoneNodeNum = 1;
+		int InletNodeNum = 4;
+		bool FirstHVACIteration = true;
+		Real64 SysMinMassFlow = 1.0 * DataEnvironment::StdRhoAir; // From inputs for Zone 1 VAV
+		Real64 SysMaxMassFlow = 2.0 * DataEnvironment::StdRhoAir; // From inputs for Zone 1 VAV
+
+		// Test with heating load
+		DataZoneEnergyDemands::ZoneSysEnergyDemand( 1 ).RemainingOutputRequired = 2000.0; // Zone 1 has a heating load - expect min flow rate
+
+		// First test - AlwaysOff Schedule - expecting no flow
+		SingleDuct::Sys( SysNum ).SchedPtr = 1;
+		DataLoopNode::Node( InletNodeNum ).MassFlowRate = SysMinMassFlow;
+		DataLoopNode::Node( InletNodeNum ).MassFlowRateMaxAvail = SysMaxMassFlow;
+		DataGlobals::BeginEnvrnFlag = true; // Must be true for initial pass thru SingleDuct::InitSys for this terminal unit
+		FirstHVACIteration = true;
+		SingleDuct::InitSys( SysNum, FirstHVACIteration ); // Run thru init once with FirstHVACIteration set to true
+		DataGlobals::BeginEnvrnFlag = false;
+		FirstHVACIteration = false;
+		SingleDuct::InitSys(SysNum, FirstHVACIteration); // Run thru init a second time with FirstHVACIteration set to false
+		SingleDuct::SimVAV( SysNum, FirstHVACIteration, ZoneNum, ZoneNodeNum );
+		EXPECT_EQ( 0.0, SingleDuct::SysOutlet( SysNum ).AirMassFlowRateMaxAvail );
+		EXPECT_EQ( 0.0, SingleDuct::SysOutlet( SysNum ).AirMassFlowRate );
```

Comment:
So if the TU is scheduled off it has 0 flow. What if there is a minimum stop > 0? I guess this means the damper goes fully closed regardless of minimum stop setting. I'm OK with this, if you want flow at minimum stop, don't schedule off the TU.